### ISSUE TITLE
Using the same value as a primary key across two different Models throws Exception

### ISSUE THREAD

**Issue by Jordan-Nelson** [aws-amplify/amplify-android#2608]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

DataStore

### Gradle script dependencies

<details>

```groovy
// Put output below this line


```

</details>

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

Originally from https://github.com/aws-amplify/amplify-flutter/issues/2519

Given the following schema

```graphql
type TestTableOne @model @auth(rules: [{allow: public}]) {
  id: ID!
  profile_id: ID! @primaryKey
  count: Int
} 

type TestTableTwo @model @auth(rules: [{allow: public}]) {
  id: ID!
  profile_id: ID! @primaryKey
  count: Int
} 

type TestTableThree @model @auth(rules: [{allow: public}]) {
  id: ID!
  profile_id: ID! @primaryKey
  count: Int
} 
```

Attempting to save a model to each of the three tables with the same `profile_id` results in an exception.

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by Jordan-Nelson** [comment]
fyi @ankpshah

**Comment by ankpshah** [comment]
This issue was fixed as part of Amplify Android Release [2.14.2](https://github.com/aws-amplify/amplify-android/releases/tag/release_v2.14.2)
